ted_sws/notice_validator/services/shacl_test_suite_runner.py

```diff
@@ -0,0 +1,104 @@
+import json
+from jinja2 import Environment, PackageLoader
+
+from ted_sws.core.model.manifestation import RDFManifestation
+from ted_sws.core.model.notice import Notice
+from ted_sws.core.model.transform import MappingSuite, SHACLTestSuite
+from ted_sws.data_manager.adapters.repository_abc import NoticeRepositoryABC, MappingSuiteRepositoryABC
+from ted_sws.notice_validator.adapters.shacl_runner import SHACLRunner
+from ted_sws.notice_validator.model.shacl_test_suite import QueriedSHACLShapeValidationResult, \
+    SHACLSuiteValidationReport
+from ted_sws.resources import SHACL_RESULT_QUERY_PATH
+
+TEMPLATES = Environment(loader=PackageLoader("ted_sws.notice_validator.resources", "templates"))
+SHACL_TEST_SUITE_EXECUTION_HTML_REPORT_TEMPLATE = "shacl_shape_validation_results_report.jinja2"
+
+
+class SHACLTestSuiteRunner:
+    """
+
+        Runs SHACL shape tests suites .
+    """
+
+    def __init__(self, rdf_manifestation: RDFManifestation, shacl_test_suite: SHACLTestSuite,
+                 mapping_suite: MappingSuite):
+        self.rdf_manifestation = rdf_manifestation
+        self.shacl_test_suite = shacl_test_suite
+        self.mapping_suite = mapping_suite
+
+    def execute_test_suite(self) -> SHACLSuiteValidationReport:
+        """
+            Executing SHACL shape validation from a SHACL test suite and return execution details
+        :return:
+        """
+        shacl_runner = SHACLRunner(self.rdf_manifestation.object_data)
+        shacl_shape_result_query = SHACL_RESULT_QUERY_PATH.read_text()
+        shacl_shape_validation_result = QueriedSHACLShapeValidationResult()
+        try:
+            shacl_shape_validation_result.identifier = self.shacl_test_suite.identifier
+            conforms, result_graph, results_text = shacl_runner.validate(self.shacl_test_suite.shacl_tests)
+            shacl_shape_validation_result.conforms = str(conforms)
+            shacl_shape_validation_result.results_dict = json.loads(
+                result_graph.query(shacl_shape_result_query).serialize(
```

Comment:
this looks rather odd, but it is a valid method to convert an object onto a nested series of dicts.